Building error on debian

I am in the process of packaging castway for debian. The build fails with an E0412:

```
  Compiling castaway v0.2.1 (/usr/share/cargo/registry/castaway-0.2.1)
     Running `CARGO=/usr/bin/cargo CARGO_CRATE_NAME=castaway CARGO_MANIFEST_DIR=/usr/share/cargo/registry/castaway-0.2.1 CARGO_PKG_AUTHORS='Stephen M. Coakley <me@stephencoakley.com>' CARGO_PKG_DESCRIPTION='Safe, zero-cost downcasting for limited compile-time specialization.' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE=MIT CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=castaway CARGO_PKG_REPOSITORY='https://github.com/sagebind/castaway' CARGO_PKG_VERSION=0.2.1 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=2 CARGO_PKG_VERSION_PATCH=1 CARGO_PKG_VERSION_PRE='' CARGO_PRIMARY_PACKAGE=1 LD_LIBRARY_PATH='/tmp/tmp.wJuFF5GI1P/target/debug/deps:/usr/lib' rustc --crate-name castaway --edition=2018 src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --emit=dep-info,link -C embed-bitcode=no -C debuginfo=2 --test -C metadata=f06c949c70501ccf -C extra-filename=-f06c949c70501ccf --out-dir /tmp/tmp.wJuFF5GI1P/target/x86_64-unknown-linux-gnu/debug/deps --target x86_64-unknown-linux-gnu -C incremental=/tmp/tmp.wJuFF5GI1P/target/x86_64-unknown-linux-gnu/debug/incremental -L dependency=/tmp/tmp.wJuFF5GI1P/target/x86_64-unknown-linux-gnu/debug/deps -L dependency=/tmp/tmp.wJuFF5GI1P/target/debug/deps --extern paste=/tmp/tmp.wJuFF5GI1P/target/debug/deps/libpaste-9b43ddbe3bcf5cdd.so --extern rustversion=/tmp/tmp.wJuFF5GI1P/target/debug/deps/librustversion-44ce5c41119c978a.so -C debuginfo=2 --cap-lints warn -C linker=x86_64-linux-gnu-gcc -C link-arg=-Wl,-z,relro --remap-path-prefix /usr/share/cargo/registry/castaway-0.2.1=/usr/share/cargo/registry/castaway-0.2.1`
error[E0433]: failed to resolve: use of undeclared type `String`
   --> src/lib.rs:482:13
    |
482 |             String::from("hello world") => Ok(ref v) if v.as_str() == "hello world",
    |             ^^^^^^ use of undeclared type `String`

error[E0412]: cannot find type `String` in this scope
   --> src/lib.rs:481:13
    |
481 |         for String {
    |             ^^^^^^ not found in this scope

Some errors have detailed explanations: E0412, E0433.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `castaway` due to 2 previous errors
```
Compiler: GCC 11

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -127,6 +127,7 @@ pub use lifetime_free::LifetimeFree;
 /// Specialization in a blanket trait implementation:
 ///
 /// ```
+/// # #[cfg(feature = "std")] {
 /// use std::fmt::Display;
 /// use castaway::cast;
 ///
@@ -160,6 +161,7 @@ pub use lifetime_free::LifetimeFree;
 ///
 /// println!("specialized: {}", String::from("hello").fast_to_string());
 /// println!("default: {}", "hello".fast_to_string());
+/// # }
 /// ```
 #[macro_export]
 macro_rules! cast {
@@ -311,6 +313,7 @@ mod tests {
 
         inner2(&slice);
 
+        #[allow(clippy::needless_lifetimes)]
         fn inner3<'a>(value: &'a mut [u8]) {
             assert_eq!(cast!(value, &mut [u8]), Ok(&mut [1, 1][..]));
         }
@@ -341,6 +344,7 @@ mod tests {
         () => {};
 
         (
+            $(#[$meta:meta])*
             for $TARGET:ty as $name:ident {
                 $(
                     $value:expr => $matches:pat $(if $guard:expr)?,
@@ -349,6 +353,7 @@ mod tests {
             $($tail:tt)*
         ) => {
             paste::paste! {
+                $(#[$meta])*
                 #[test]
                 #[allow(non_snake_case)]
                 fn [<cast_lifetime_free_ $name>]() {
@@ -364,6 +369,7 @@ mod tests {
                     )*
                 }
 
+                $(#[$meta])*
                 #[test]
                 #[allow(non_snake_case)]
                 fn [<cast_lifetime_free_ref_ $name>]() {
@@ -379,6 +385,7 @@ mod tests {
                     )*
                 }
 
+                $(#[$meta])*
                 #[test]
                 #[allow(non_snake_case)]
                 fn [<cast_lifetime_free_mut_ $name>]() {
@@ -401,6 +408,7 @@ mod tests {
         };
 
         (
+            $(#[$meta:meta])*
             for $TARGET:ty {
                 $(
                     $value:expr => $matches:pat $(if $guard:expr)?,
@@ -409,6 +417,7 @@ mod tests {
             $($tail:tt)*
         ) => {
             paste::paste! {
+                $(#[$meta])*
                 #[test]
                 #[allow(non_snake_case)]
                 fn [<cast_lifetime_free_ $TARGET>]() {
@@ -424,6 +433,7 @@ mod tests {
                     )*
                 }
 
+                $(#[$meta])*
                 #[test]
                 #[allow(non_snake_case)]
                 fn [<cast_lifetime_free_ref_ $TARGET>]() {
@@ -439,6 +449,7 @@ mod tests {
                     )*
                 }
 
+                $(#[$meta])*
                 #[test]
                 #[allow(non_snake_case)]
                 fn [<cast_lifetime_free_mut_ $TARGET>]() {
@@ -478,6 +489,7 @@ mod tests {
             3.2f64 => Err(v) if v == 3.2f64,
         }
 
+        #[cfg(feature = "std")]
         for String {
             String::from("hello world") => Ok(ref v) if v.as_str() == "hello world",
             "hello world" => Err("hello world"),

